### PR TITLE
Add get_newest_ping_per_client function

### PR DESCRIPTION
**Pull request by poiru**

### CODE REVIEW

File: moztelemetry/spark.py [hunk 1]
```diff
@@ -133,22 +133,18 @@ def get_one_ping_per_client(pings):
...
-                    map(lambda p: p[1])
+def get_newest_ping_per_client(pings):
+    """
+    Returns the newest ping of each client in the RDD based on the submission
+    date. The caveats of get_one_ping_per_client apply here as well.
+    """
+    submission_date = "meta/submissionDate"
+    reduce_func = lambda p1, p2:\
+        p1 if p1[submission_date] < p2[submission_date] else p2
+    return _get_one_ping_per_client(pings, reduce_func)
```
Comment (vitillo) [inline moztelemetry/spark.py]: Isn't this returing the oldest ping instead of the newest one?

Comment (poiru) [inline moztelemetry/spark.py]: @vitillo Oops, yes! Fixed.

**Comment by vitillo** [comment]
Looks good, thanks.

**Comment by mreid-moz** [comment]
It would be nice to modify this to add a "ping_selector" function as a parameter to `get_one_ping_per_client`, defaulting to the current `_get_one_ping_per_client`, but letting you do whatever you need - then it would be easy to implement "get newest", "get oldest", "get longest duration" etc.

**Comment by vitillo** [comment]
+1

**Comment by poiru** [comment]
@mreid-moz Good idea, submitted PR #25!

**Pull request merged**